ft_denoise_hfc.m

```diff
@@ -0,0 +1,240 @@
+%   cfg.channel         = channels for HFC (default = 'all')
+% 	cfg.L               = spherical harmonic order, L=1 is a homogenous field.
+%                           L=2 includes gradients, L=3 quadratic terms etc.
+%   cfg.trials          = which trials do you want to denoise?
+%                           (default = 'all')
+%   cfg.updatesens      = do you want to update sensor info with projector?
+%                           (default = 'yes')
+%   cfg.feedback        = do you want feedback (default = 'yes')
+%   cfg.residualcheck   = do you want to check channel residuals
+%                          (default = 'yes')
+%   cfg.residualthresh  = (in pT) what level of residual signal is fine for
+%                           quality assurance (default=50)
+% See also FT_DENOISE_SYNTHETIC, FT_DENOISE_PCA, FT_DENOISE_DSSP, FT_DENOISE_TSP
+
+% Copyright (C) 2021-22, Tim Tierney, George O'Neill, Robert Seymour
+% Wellcome Centre for Human Neuroimaging, UCL
+%
+% This file is part of FieldTrip, see http://www.fieldtriptoolbox.org
+% for the documentation and details.
+%
+%    FieldTrip is free software: you can redistribute it and/or modify
+%    it under the terms of the GNU General Public License as published by
+%    the Free Software Foundation, either version 3 of the License, or
+%    (at your option) any later version.
+%
+%    FieldTrip is distributed in the hope that it will be useful,
+%    but WITHOUT ANY WARRANTY; without even the implied warranty of
+%    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%    GNU General Public License for more details.
+%
+%    You should have received a copy of the GNU General Public License
+%    along with FieldTrip. If not, see <http://www.gnu.org/licenses/>.
+%
+% $Id$
+
+% these are used by the ft_preamble/ft_postamble function and scripts
+ft_revision = '$Id$';
+ft_nargin   = nargin;
+ft_nargout  = nargout;
+
+% do the general setup of the function
+ft_defaults
+ft_preamble init
+ft_preamble debug
+ft_preamble loadvar    data
+ft_preamble provenance data
+ft_preamble trackconfig
+
+% the ft_abort variable is set to true or false in ft_preamble_init
+if ft_abort
+    % do not continue function execution in case the outputfile is present and the user indicated to keep it
+    return
+end
+
+% check if the input cfg is valid for this function
+cfg = ft_checkconfig(cfg, 'forbidden',  {'trial'}); % prevent accidental typos, see issue 1729
+
+% check the input data
+data = ft_checkdata(data, 'datatype', {'raw'});
+
+% check if the input cfg is valid for this function
+cfg = ft_checkconfig(cfg, 'forbidden',  {'trial'}); % prevent accidental typos, see issue 1729
+cfg = ft_checkconfig(cfg, 'forbidden',  {'channels'}); % prevent accidental typos, see issue 1729
+
+% Check if the data has a grad structure
+if ~isfield(data,'grad')
+    error(['Data needs a grad structure']);
+end
+
+% set the defaults
+cfg.L               = ft_getopt(cfg, 'L', 1);
```

Comment:
for consideration: may I suggest to change the option cfg.L into cfg.order? This makes the meaning of the option a bit more intuitive to the user.